server/gokbg3/grails-app/services/org/gokb/ComponentLookupService.groovy

```diff
@@ -639,4 +641,27 @@ class ComponentLookupService {
 
     return result
   }
+
+
+  CuratoryGroup findCuratoryGroupOfInterest(@Nonnull KBComponent component, User user = null){
+    // TODO: to be extended for further comparision objects
+    if (component.curatoryGroups?.size() == 1){
```

Comment:
TitleInstances for example don't have the Combo property `curatoryGroups`, therfore this throws a MissingPropertyException. The best way to check is `KBComponent.has(object, 'curatoryGroups')`.